Volume/Supporting Views/ReaderToolbarView.swift

```diff
@@ -0,0 +1,230 @@
+//
+//  ReaderToolbarView.swift
+//  Volume
+//
+//  Created by Hanzheng Li on 11/30/22.
+//  Copyright © 2022 Cornell AppDev. All rights reserved.
+//
+
+import AppDevAnalytics
+import Combine
+import SDWebImageSwiftUI
+import SwiftUI
+
+struct ReaderToolbarView<Content: ReadableContent>: View {
+
+    let content: Content?
+    let navigationSource: NavigationSource
+
+    @EnvironmentObject private var userData: UserData
+    @State private var bookmarkRequestInProgress: Bool = false
+    @State private var queryBag = Set<AnyCancellable>()
+
+    // MARK: Data
+
+    private func incrementShoutouts(for content: Content) {
+        guard let uuid = userData.uuid else {
+            return
+        }
+
+        if let article = content as? Article {
```

Comment:
Can we clean up this code using guard let and helper functions, it's very hard to follow